tests/functional/adapter/test_backup_table.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from dbt.tests.util import run_dbt
+
+
+_MODELS = {
+    "backup_is_false.sql": "{{ config(materialized='table', backup=False) }} select 1 as my_col",
+    "backup_is_true.sql": "{{ config(materialized='table', backup=True) }} select 1 as my_col",
+    "backup_is_undefined.sql": "{{ config(materialized='table') }} select 1 as my_col",
+    "backup_is_true_view.sql": "{{ config(materialized='view', backup=True) }} select 1 as my_col",
+    "syntax_with_distkey.sql": "{{ config(materialized='table', backup=False, dist='my_col') }} select 1 as my_col",
+    "syntax_with_sortkey.sql": "{{ config(materialized='table', backup=False, sort='my_col') }} select 1 as my_col",
+}
+
+
+class BackupTableBase:
+
+    @pytest.fixture(scope="class", autouse=True)
+    def run_dbt_results(self, project):
+        yield run_dbt(["run"])
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return _MODELS
+
+
+class TestBackupTableSetup(BackupTableBase):
+
+    def test_setup_executed_correctly(self, run_dbt_results):
```

Comment:
Why have this as a test? Is it good to have tests test the test itself (lol, wording) or should that be an assert? Curious what you have to say.

Reply:
I put this here because the integration test was effectively doing this for each test. So I removed that piece and ran it once instead of n times. It also helped me make sure that I was setting this up correctly as this was my first conversion. I agree it can be removed.